sdk/v1/client.go

```diff
@@ -114,17 +134,23 @@ func (cli *Client) callReindexAPI(ctx context.Context, path, method string, head
 
 	// check req, above, didn't error
 	if err != nil {
-		return nil, apiError.StatusError{
+		return nil, nil, apiError.StatusError{
 			Err:  fmt.Errorf("failed to create request for call to search reindex api, error is: %v", err),
 			Code: http.StatusInternalServerError,
 		}
 	}
 
-	headers.Add(req)
+	err = headers.Add(req)
+	if err != nil {
+		return nil, nil, apiError.StatusError{
+			Err:  fmt.Errorf("failed to add headers to request, error is: %v", err),
```

Comment:
log rather than fmt, I think

Reply:
I don't think I can use log here as it would return a string but it needs to be a value for the Err in the apiError.StatusError struct. I would prefer to leave it as it is. This code is only changed because I added a nil to the values being returned in line 145.